File: next/security/user and permission/backend-ms1/src/main/java/ma/zs/stocky/ws/facade/admin/security/ModelPermissionUserRestAdmin.java
package  ma.zs.stocky.ws.facade.admin.security;

import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.media.Content;
import io.swagger.v3.oas.annotations.media.Schema;
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import io.swagger.v3.oas.annotations.responses.ApiResponses;

import ma.zs.stocky.bean.core.security.ModelPermissionUser;
import ma.zs.stocky.dao.criteria.core.security.ModelPermissionUserCriteria;
import ma.zs.stocky.service.facade.admin.security.ModelPermissionUserAdminService;
import ma.zs.stocky.ws.converter.security.ModelPermissionUserConverter;
import ma.zs.stocky.ws.dto.security.ModelPermissionUserDto;
import ma.zs.stocky.zynerator.controller.AbstractController;
import ma.zs.stocky.zynerator.dto.AuditEntityDto;
import ma.zs.stocky.zynerator.util.PaginatedList;


import org.springframework.core.io.InputStreamResource;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;


import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import ma.zs.stocky.zynerator.process.Result;


import org.springframework.web.multipart.MultipartFile;
import ma.zs.stocky.zynerator.dto.FileTempDto;

@RestController
@RequestMapping("/api/admin/modelPermissionUser/")
public class ModelPermissionUserRestAdmin  extends AbstractController<ModelPermissionUser, ModelPermissionUserDto, ModelPermissionUserCriteria, ModelPermissionUserAdminService, ModelPermissionUserConverter> {



    @Operation(summary = "upload one modelPermissionUser")
    @RequestMapping(value = "upload", method = RequestMethod.POST, consumes = "multipart/form-data")
    public ResponseEntity<FileTempDto> uploadFileAndGetChecksum(@RequestBody MultipartFile file) throws Exception {
        return super.uploadFileAndGetChecksum(file);
    }
    @Operation(summary = "upload multiple modelPermissionUsers")
    @RequestMapping(value = "upload-multiple", method = RequestMethod.POST, consumes = "multipart/form-data")
    public ResponseEntity<List<FileTempDto>> uploadMultipleFileAndGetChecksum(@RequestBody MultipartFile[] files) throws Exception {
        return super.uploadMultipleFileAndGetChecksum(files);
    }

    @Operation(summary = "Finds a list of all modelPermissionUsers")
    @GetMapping("")
    public ResponseEntity<List<ModelPermissionUserDto>> findAll() throws Exception {
        return super.findAll();
    }



    @Operation(summary = "Saves the specified  modelPermissionUser")
    @PostMapping("")
    public ResponseEntity<ModelPermissionUserDto> save(@RequestBody ModelPermissionUserDto dto) throws Exception {
        return super.save(dto);
    }

    @Operation(summary = "Updates the specified  modelPermissionUser")
    @PutMapping("")
    public ResponseEntity<ModelPermissionUserDto> update(@RequestBody ModelPermissionUserDto dto) throws Exception {
        return super.update(dto);
    }

    @Operation(summary = "Delete list of modelPermissionUser")
    @PostMapping("multiple")
    public ResponseEntity<List<ModelPermissionUserDto>> delete(@RequestBody List<ModelPermissionUserDto> listToDelete) throws Exception {
        return super.delete(listToDelete);
    }
    @Operation(summary = "Delete the specified modelPermissionUser")
    @DeleteMapping("")
    public ResponseEntity<ModelPermissionUserDto> delete(@RequestBody ModelPermissionUserDto dto) throws Exception {
            return super.delete(dto);
    }

    @Operation(summary = "Delete the specified modelPermissionUser")
    @DeleteMapping("id/{id}")
    public ResponseEntity<Long> deleteById(@PathVariable Long id) throws Exception {
        return super.deleteById(id);
    }
    @Operation(summary = "Delete multiple modelPermissionUsers by ids")
    @DeleteMapping("multiple/id")
    public ResponseEntity<List<Long>> deleteByIdIn(@RequestBody List<Long> ids) throws Exception {
            return super.deleteByIdIn(ids);
     }


    @Operation(summary = "find by actionPermission id")
    @GetMapping("actionPermission/id/{id}")
    public List<ModelPermissionUserDto> findByActionPermissionId(@PathVariable Long id){
        return findDtos(service.findByActionPermissionId(id));
    }
    @Operation(summary = "delete by actionPermission id")
    @DeleteMapping("actionPermission/id/{id}")
    public int deleteByActionPermissionId(@PathVariable Long id){
        return service.deleteByActionPermissionId(id);
    }
    @Operation(summary = "find by modelPermission id")
    @GetMapping("modelPermission/id/{id}")
    public List<ModelPermissionUserDto> findByModelPermissionId(@PathVariable Long id){
        return findDtos(service.findByModelPermissionId(id));
    }
    @Operation(summary = "delete by modelPermission id")
    @DeleteMapping("modelPermission/id/{id}")
    public int deleteByModelPermissionId(@PathVariable Long id){
        return service.deleteByModelPermissionId(id);
    }
    @Operation(summary = "find by user id")
    @GetMapping("user/id/{id}")
    public List<ModelPermissionUserDto> findByUserId(@PathVariable Long id){
        return findDtos(service.findByUserId(id));
    }
    @Operation(summary = "delete by user id")
    @DeleteMapping("user/id/{id}")
    public int deleteByUserId(@PathVariable Long id){
        return service.deleteByUserId(id);
    }

    @Operation(summary = "Finds a modelPermissionUser and associated list by id")
    @GetMapping("id/{id}")
    public ResponseEntity<ModelPermissionUserDto> findById(@PathVariable Long id) {
        return super.findById(id);
    }

    @Operation(summary = "Finds modelPermissionUsers by criteria")
    @PostMapping("find-by-criteria")
    public ResponseEntity<List<ModelPermissionUserDto>> findByCriteria(@RequestBody ModelPermissionUserCriteria criteria) throws Exception {
        return super.findByCriteria(criteria);
    }

    @Operation(summary = "Finds paginated modelPermissionUsers by criteria")
    @PostMapping("find-paginated-by-criteria")
    public ResponseEntity<PaginatedList> findPaginatedByCriteria(@RequestBody ModelPermissionUserCriteria criteria) throws Exception {
        return super.findPaginatedByCriteria(criteria);
    }

    @Operation(summary = "Exports modelPermissionUsers by criteria")
    @PostMapping("export")
    public ResponseEntity<InputStreamResource> export(@RequestBody ModelPermissionUserCriteria criteria) throws Exception {
        return super.export(criteria);
    }

    @Operation(summary = "Gets modelPermissionUser data size by criteria")
    @PostMapping("data-size-by-criteria")
    public ResponseEntity<Integer> getDataSize(@RequestBody ModelPermissionUserCriteria criteria) throws Exception {
        return super.getDataSize(criteria);
    }



    public ModelPermissionUserRestAdmin (ModelPermissionUserAdminService service, ModelPermissionUserConverter converter) {
        super(service, converter);
    }




}

File: nest-angular/security-v2/backend-ms1/src/app/module/ws/admin/security/ModelPermissionUserRestAdmin.ts
import {Body, Controller,Delete , Get, Param, Post, Put} from '@nestjs/common';
import {ApiOperation, ApiTags} from '@nestjs/swagger';
import {PaginatedList} from "src/app/zynerator/util/PaginatedList";

import {ModelPermissionUserConverter} from "src/app/shared/converter/security/ModelPermissionUserConverter";
import {ModelPermissionUserDto} from "src/app/shared/dto/security/ModelPermissionUserDto";
import {ModelPermissionUserCriteria} from "src/app/shared/dao/criteria/core/security/ModelPermissionUserCriteria";
import {ModelPermissionUserAdminServiceImpl} from "src/app/module/service/admin/security/ModelPermissionUserAdminServiceImpl";


@ApiTags('Manages model permission user services')
@Controller('api/admin/modelPermissionUser')
export class ModelPermissionUserRestAdmin {

    constructor(private readonly service: ModelPermissionUserAdminServiceImpl,
                private readonly converter: ModelPermissionUserConverter) {
    }

    @ApiOperation({summary: 'Finds a list of all modelPermissionUsers'})
    @Get()
    async findAll(): Promise<ModelPermissionUserDto[]> {
        const items = await this.service.findAll();
        return this.converter.toDtos(items);
    }

    @ApiOperation({summary: 'Finds a model permission user by id'})
    @Get('id/:id')
    async findById(@Param('id') id: number): Promise<ModelPermissionUserDto> {
        const item = await this.service.findById(id);
        return this.converter.toDto(item);
    }

    @ApiOperation({summary: 'Deletes a model permission user by id'})
    @Delete('id/:id')
    async deleteById(@Param('id') id: number): Promise<number> {
        return this.service.deleteById(id);
    }

    @ApiOperation({summary: 'Saves the specified model permission user'})
    @Post()
    async save(@Body() dto: ModelPermissionUserDto): Promise<ModelPermissionUserDto> {
        const item = this.converter.toItem(dto);
        const savedItem = await this.service.save(item);
        return this.converter.toDto(savedItem);
    }


    @ApiOperation({summary: 'Updates the specified model permission user'})
    @Put()
    async update(@Body() dto: ModelPermissionUserDto): Promise<ModelPermissionUserDto> {
        const item = this.converter.toItem(dto);
        const result = await this.service.updateWithAssociatedLists(item);
        return this.converter.toDto(result);
   }

    @ApiOperation({summary: 'Finds an optimized list of all modelPermissionUsers'})
    @Get('optimized')
    async findAllOptimized(): Promise<ModelPermissionUserDto[]> {
        const result = await this.service.findAllOptimized();
        return result;
    }

    @ApiOperation({summary: 'Finds an optimized list of all modelPermissionUsers'})
    @Post('find-by-criteria')
    async findByCriteria(@Body() criteria: ModelPermissionUserCriteria): Promise<ModelPermissionUserDto[]> {
        const items = await this.service.findByCriteria(criteria);
        return this.converter.toDtos(items);
    }

    @ApiOperation({summary: 'Finds an optimized list of all modelPermissionUsers'})
    @Post('find-paginated-by-criteria')
    async findPaginatedByCriteria(@Body() criteria: ModelPermissionUserCriteria): Promise<PaginatedList<ModelPermissionUserDto>> {
        const paginated = await this.service.findPaginatedByCriteria(criteria);
        const dtos = this.converter.toDtos(paginated.list);
        return new PaginatedList<ModelPermissionUserDto>(dtos, paginated.dataSize);
    }


    @Get('actionPermission/id/:id')
    async findByActionPermissionId(@Param('id') id: number): Promise<ModelPermissionUserDto[]> {
        const items = await this.service.findByActionPermissionId(id);
        return this.converter.toDtos(items);
    }
    @Get('modelPermission/id/:id')
    async findByModelPermissionId(@Param('id') id: number): Promise<ModelPermissionUserDto[]> {
        const items = await this.service.findByModelPermissionId(id);
        return this.converter.toDtos(items);
    }
    @Get('user/id/:id')
    async findByUserId(@Param('id') id: number): Promise<ModelPermissionUserDto[]> {
        const items = await this.service.findByUserId(id);
        return this.converter.toDtos(items);
    }

    @Post('multiple')
    async deleteMultiple(@Body() dtos: ModelPermissionUserDto[]): Promise<ModelPermissionUserDto[]> {
        const items = dtos.map(dto => this.converter.toItem(dto));
        const deletedItems = await this.service.deleteMultiple(items);
        return deletedItems.map(element => this.converter.toDto(element));
    }

}
